exchanges/bybit/bybit_websocket.go

```diff
@@ -107,443 +113,645 @@ func (by *Bybit) WsAuth(ctx context.Context) error {
+
+	pairFormat, err := by.GetPairFormat(assetType, true)
+	if err != nil {
+		return nil, err
+	}
+	for i := range channelsToSubscribe {
+		switch channelsToSubscribe[i].Channel {
+		case chanOrderbook:
+			arg.Arguments = append(arg.Arguments, fmt.Sprintf("%s.%d.%s", channelsToSubscribe[i].Channel, 50, channelsToSubscribe[i].Currency.Format(pairFormat).String()))
+		case chanPublicTrade, chanPublicTicker, chanLiquidation, chanLeverageTokenTicker, chanLeverageTokenNav:
+			arg.Arguments = append(arg.Arguments, channelsToSubscribe[i].Channel+"."+channelsToSubscribe[i].Currency.Format(pairFormat).String())
+		case chanKline, chanLeverageTokenKline:
+			interval, err := intervalToString(kline.FiveMin)
+			if err != nil {
+				return nil, err
 			}
-		} else {
-			subReq.Parameters = WsParams{
-				Symbol:   formattedPair.String(),
-				IsBinary: true,
+			arg.Arguments = append(arg.Arguments, channelsToSubscribe[i].Channel+"."+interval+"."+channelsToSubscribe[i].Currency.Format(pairFormat).String())
+		case chanPositions, chanExecution, chanOrder, chanWallet, chanGreeks, chanDCP:
+			if chanMap[channelsToSubscribe[i].Channel]|selectedChannels > 0 {
+				continue
 			}
+			authArg.Arguments = append(authArg.Arguments, channelsToSubscribe[i].Channel)
+			selectedChannels |= chanMap[channelsToSubscribe[i].Channel]
 		}
-		err = by.Websocket.Conn.SendJSONMessage(subReq)
-		if err != nil {
-			errs = common.AppendError(errs, err)
-			continue
+		if len(arg.Arguments) >= 10 {
+			args = append(args, arg)
+			arg = SubscriptionArgument{
+				Operation: operation,
+				RequestID: by.Websocket.Conn.GenerateMessageID(false),
+				Arguments: []string{},
+			}
 		}
-		by.Websocket.AddSuccessfulSubscriptions(channelsToSubscribe[i])
 	}
-	return errs
+	if len(arg.Arguments) != 0 {
+		args = append(args, arg)
+	}
+	if len(authArg.Arguments) != 0 {
+		args = append(args, authArg)
+	}
+	return args, nil
 }
 
 // Unsubscribe sends a websocket message to stop receiving data from the channel
 func (by *Bybit) Unsubscribe(channelsToUnsubscribe []stream.ChannelSubscription) error {
-	var errs error
-
-	for i := range channelsToUnsubscribe {
-		var unSub WsReq
-		unSub.Event = cancel
-		unSub.Topic = channelsToUnsubscribe[i].Channel
+	return by.handleSpotSubscription("unsubscribe", channelsToUnsubscribe)
+}
 
-		formattedPair, err := by.FormatExchangeCurrency(channelsToUnsubscribe[i].Currency, asset.Spot)
-		if err != nil {
-			errs = common.AppendError(errs, err)
-			continue
-		}
-		unSub.Parameters = WsParams{
-			Symbol: formattedPair.String(),
+func (by *Bybit) handleSpotSubscription(operation string, channelsToSubscribe []stream.ChannelSubscription) error {
+	payloads, err := by.handleSubscriptions(asset.Spot, operation, channelsToSubscribe)
+	if err != nil {
+		return err
+	}
+	for a := range payloads {
+		var response []byte
+		if payloads[a].auth {
+			response, err = by.Websocket.AuthConn.SendMessageReturnResponse(payloads[a].RequestID, payloads[a])
```

Comment:
ReqestID is an int64, while a string is returned for matching; perhaps make this a string?